runtime/riscv.S

```diff
@@ -362,6 +518,23 @@ L(caml_c_call):
         ld      ALLOC_PTR, Caml_state(young_ptr)
     /* Load ocaml stack */
         SWITCH_C_TO_OCAML
+#if defined(WITH_THREAD_SANITIZER)
+    /* Save return value registers. Since the called function could be
+       anything, it may have returned its result (if any) either in a0
+       or fa0:fa1. */
+        addi    sp, sp, -32
+        CFI_ADJUST(32)
+        sd      a0, 0(sp)
```

Comment:
`caml_c_call` wraps any kind of C function, and needs to preserve the returned value during the TSan instrumentation.

On RiscV, the ABI requires that integer values are returned into `a0`, or `a0:a1` if the return value does not fit in one register, and that floating-point values are similarly returned in `fa0` or `fa0:fa1`.

On 64-bit RiscV, until C grows a 128-bit integer type, all integer types will fit in a single register, so we only need to save `a0`. However, a `long double` return type would span both `fa0` and `fa1`, so we save them both.

Reply:
I was checking with @OlivierNicole our choice not to save `xmm1` in `amd64.S`: the reasoning was that the compatible returned type of the C function called by OCaml would be only be `float` (`double` in C) or `int32`/`int64`/`nativeint` (all fitting in 64bits register), so we don't bother saving `rdx` and `xmm1`. See the table in <https://v2.ocaml.org/manual/intfc.html#s:C-cheaper-call> for reference.
Do you agree with this, and if so, pehaps we shouldn't save `fa1` either to be consistent with `amd64.S`.

Reply:
Given that limited list of supported types, I agree that only one floating-point register needs to be saved here. Since the arm64 code is already in and saves two, I intend to clean this in a later PR once all TSan support PRs are in (since all of them save two floating-point registers here), if that's ok with you.